src/bplus_tree.rs

```diff
@@ -0,0 +1,525 @@
+                } else {
+                    node.children[i] = Some(temp);
+                }
+            }
+        }
+    }
+
+    fn merge(&self, node1: &mut Box<Node<K>>, node2: &mut Box<Node<K>>) {
+        let new_key_num = node1.key_num + node2.key_num;
+
+        node1.keys.resize(new_key_num, K::default());
+        if node1.leaf {
+            node1.pointers.resize(new_key_num, ChunkHandler::default());
+        } else {
+            node1.children.resize(new_key_num + 2, None);
+        }
+        for i in 0..node2.key_num {
+            let temp = node1.key_num;
+            node1.keys[temp + i] = node2.keys[i].clone();
+            if node1.leaf {
+                node1.pointers[temp] = node2.pointers[i].clone();
+            } else {
+                node1.children[temp + i + 1] = node2.children[i].clone();
+            }
+        }
+
+        if !node1.leaf {
+            node1.children[new_key_num + 1] = node2.children[node2.key_num].clone();
+        }
+    }
+}
+
+impl<K: Ord + Clone + Default> Database<K, DataContainer<()>> for BPlus<K> {
+    fn insert(&mut self, key: K, value: DataContainer<()>) -> io::Result<()> {
+        match value.extract() {
+            Data::Chunk(chunk) => self.insert(key, chunk.clone()),
+            Data::TargetChunk(_chunk) => unimplemented!(),
+        }
+    }
+
+    fn get(&self, key: &K) -> io::Result<DataContainer<()>> {
+        match self.get(key) {
+            Ok(chunk) => Ok(DataContainer::from(chunk)),
+            Err(error) => Err(error),
+        }
+    }
+
+    fn contains(&self, key: &K) -> bool {
+        self.contains(key)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashMap;
+
+    use tempdir::TempDir;
+
+    use super::*;
+
+    #[test]
+    fn test_insert_and_find() {
+        let tempdir = &TempDir::new("1").unwrap();
+        let path = PathBuf::new().join(tempdir.path());
+        let mut tree: BPlus<usize> = BPlus::new(2, path);
+        let _ = tree.insert(1, vec![1 as u8; 1]);
+        let _ = tree.insert(2, vec![2 as u8; 1]);
+        let _ = tree.insert(3, vec![2 as u8; 1]);
+        let _ = tree.insert(4, vec![2 as u8; 1]);
+        let _ = tree.insert(5, vec![2 as u8; 1]);
+        let a = tree.get(&1).unwrap();
+        assert_eq!(a, vec![1 as u8; 1]);
+    }
+
+    #[test]
+    fn test_insert_and_delete() {
+        let tempdir = &TempDir::new("2").unwrap();
+        let path = PathBuf::new().join(tempdir.path());
+        let mut tree: BPlus<usize> = BPlus::new(2, path);
+        let _ = tree.insert(1, vec![1 as u8; 1]);
+        let _ = tree.insert(2, vec![2 as u8; 1]);
+        let _ = tree.insert(3, vec![2 as u8; 1]);
+        let _ = tree.insert(4, vec![2 as u8; 1]);
+        let _ = tree.insert(5, vec![2 as u8; 1]);
+        tree.remove(&1);
+        let a = tree.get(&1);
+        match a {
+            Ok(_x) => assert!(false),
+            Err(_x) => return,
+        }
```

Comment:
```suggestion
        let a = tree.get(&1);
        assert!(a.is_err())
```
можно еще ErrorKind проверить